DataTypes.h

```diff
@@ -1,12 +1,26 @@
 #ifndef DEVICE_DATA_TYPES_H
 #define DEVICE_DATA_TYPES_H
 
+#include <cstddef>
+#include <limits>
+
 #if REAL_SIZE == 8
-typedef double real;
+using real = double;
 #elif REAL_SIZE == 4
-typedef float real;
+using real = float;
 #else
 #error REAL_SIZE not supported.
 #endif
 
+namespace device {
+struct DeviceGraphHandle {
+  bool isInitialized() const {
+    return graphID != std::numeric_limits<size_t>::max();
+  }
+  bool operator!() const {
```

Comment:
IMO this should also implement operator bool() to not be confusing

Reply:
done

https://github.com/SeisSol/Device/blob/ae2a6baad226ec54bf0bca9da0145f91cca19a3d/DataTypes.h#L29-L31